src/group_theory/subgroup.lean

```diff
@@ -1773,6 +1773,22 @@ lemma mem_gpowers_iff {g h : G} :
   h ∈ gpowers g ↔ ∃ (k : ℤ), g ^ k = h :=
 iff.rfl
 
+@[simp] lemma forall_gpowers {x : G} {p : gpowers x → Prop} :
+  (∀ g, p g) ↔ ∀ m : ℤ, p ⟨x ^ m, m, rfl⟩ :=
+⟨λ H m, H _, λ H ⟨y, m, hy⟩, by { subst y, apply H }⟩
+
+@[simp] lemma exists_gpowers {x : G} {p : gpowers x → Prop} :
+  (∃ g, p g) ↔ ∃ m : ℤ, p ⟨x ^ m, m, rfl⟩ :=
+⟨λ ⟨⟨y, m, hm⟩, hy⟩, ⟨m, by { subst y, exact hy }⟩, λ ⟨m, h⟩, ⟨_, h⟩⟩
```

Comment:
This feels like it should have some existing lemma about `coe_sort (set.range _)` that we could use

Reply:
That is, something like:
```lean
theorem forall_range_subtype_iff {α β} (f : β → α) {p : set.range f → Prop} :
  (∀ a : set.range f, p a) ↔ (∀ b, p ⟨f b, b, rfl⟩) :=
subtype.forall.trans ⟨λ h b, h (f b) _, λ h a ⟨b, hb⟩, by { convert h b, exact hb.symm }⟩
```